lib/octokit/client/stats.rb

```diff
@@ -75,11 +75,26 @@ def punch_card_stats(repo, options = {})
       #
       # @param repo [Integer, String, Hash, Repository] A GitHub repository
       # @param metric [String] The metrics you are looking for
-      # @return [Array<Sawyer::Resource>] Magical unicorn stats
+      # @return [Array<Sawyer::Resource> or nil] Magical unicorn stats or nil if unicorn haven't calculated stats yet
       def get_stats(repo, metric, options = {})
-        data = get("#{Repository.path repo}/stats/#{metric}", options)
+        get_stats_data_patiently(repo, metric, options).tap do
+          return nil if last_response.status == 202
+        end
+      end
+
+      # @private Get stats for a repository. Repeat call to Github until stats are ready or [Configurable#stats_timeout] exceeded.
+      #
+      # @param repo [Integer, String, Hash, Repository] A GitHub repository
+      # @param metric [String] The metrics you are looking for
+      # @return [Array<Sawyer::Resource>] Magical unicorn stats
+      def get_stats_data_patiently(repo, metric, options = {})
+        time_start = Time.now
 
-        last_response.status == 202 ? nil : data
+        loop do
+          data = get("#{Repository.path repo}/stats/#{metric}", options)
+          return data if last_response.status == 200 || Time.now - time_start >= stats_timeout
+          sleep 0.1
```

Comment:
Delay between retries could be an issue here. Even if the API takes a few seconds to generate the stats, at `0.1` this could add up to a lot of wasted requests. It should probably be longer. Not to make it too complicated, but maybe it could delay longer and longer in a number of steps until it reaches the set timeout.